### PR TITLE
Fix ONNX export for MobileViT for segmentation

### LINKED ISSUE #1127: ValueError: Unrecognized configuration class <class 'transformers.models.mobilevit.configuration_mobilevit.MobileViTConfig'> for this kind of AutoModel: AutoModelForImageSegmentation.
### System Info

```shell
- `optimum` version: 1.8.8.dev0
- `transformers` version: 4.30.2
- Platform: Linux-5.15.107+-x86_64-with-glibc2.31
- Python version: 3.10.12
- Huggingface_hub version: 0.15.1
- PyTorch version (GPU?): 2.0.1+cu118 (cuda availabe: False)
- Tensorflow version (GPU?): not installed (cuda availabe: NA)
```

### Who can help?

@michaelbenayoun 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

Running `optimum-cli export onnx -m apple/deeplabv3-mobilevit-small out`

results in:
```
Framework not specified. Using pt to export to ONNX.
Traceback (most recent call last):
  File "/usr/local/bin/optimum-cli", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/dist-packages/optimum/commands/optimum_cli.py", line 163, in main
    service.run()
  File "/usr/local/lib/python3.10/dist-packages/optimum/commands/export/onnx.py", line 219, in run
    main_export(
  File "/usr/local/lib/python3.10/dist-packages/optimum/exporters/onnx/__main__.py", line 184, in main_export
    model = TasksManager.get_model_from_task(
  File "/usr/local/lib/python3.10/dist-packages/optimum/exporters/tasks.py", line 1470, in get_model_from_task
    model = model_class.from_pretrained(model_name_or_path, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/transformers/models/auto/auto_factory.py", line 487, in from_pretrained
    raise ValueError(
ValueError: Unrecognized configuration class <class 'transformers.models.mobilevit.configuration_mobilevit.MobileViTConfig'> for this kind of AutoModel: AutoModelForImageSegmentation.
Model type should be one of DetrConfig.
```

### Expected behavior

The model should export correctly.

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

Fixes #1127. In the `TasksManager`, `image-segmentation` should also be associated to `AutoModelForSemanticSegmentation` (see https://github.com/huggingface/transformers/blob/2c977e4a9031e3961ef185b40ff4b7f54eda363c/src/transformers/pipelines/__init__.py#L345).

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by xenova** [comment]
Can confirm the models exported correctly! 👍 
- https://huggingface.co/Xenova/deeplabv3-mobilevit-small
- https://huggingface.co/Xenova/deeplabv3-mobilevit-x-small
- https://huggingface.co/Xenova/deeplabv3-mobilevit-xx-small

**Comment by fxmarty** [comment]
To me it feels more like the models on the Hub are wrongly labeled as `image-segmentation`.

**Comment by regisss** [comment]
> To me it feels more like the models on the Hub are wrongly labeled as `image-segmentation`.

Well, semantic segmentation is one type of image segmentation so it kind of make sense, even though I don't know why there are no fine-grained labels for segmentation. But the fact is that there is only one segmentation task on the hub when searching models and it is image segmentation: https://huggingface.co/models?pipeline_tag=image-segmentation&sort=downloads
That means that all semantic segmentation models are tagged that way. Changing this would entail changes in Transformers and maybe other libs, so that's not really an option.

File: optimum/exporters/onnx/model_configs.py [hunk 1]
```diff
@@ -567,7 +568,14 @@ class ConvNextOnnxConfig(ViTOnnxConfig):
 
 
 class MobileViTOnnxConfig(ViTOnnxConfig):
-    pass
+    ATOL_FOR_VALIDATION = 1e-4
+
+    @property
+    def outputs(self) -> Dict[str, Dict[int, str]]:
+        if self.task == "image-segmentation":
+            return copy.deepcopy(self._TASK_TO_COMMON_OUTPUTS["semantic-segmentation"])
```
Comment (michaelbenayoun) [inline optimum/exporters/onnx/model_configs.py:576]: Are `semantic-segmentation` a synonym to `image-segmentation`?

If so I would rather use the "synonyms" system we have in the `TasksManager`.

Comment (regisss) [reply optimum/exporters/onnx/model_configs.py:576]: I don't think they are. The thing is that `AutoModelForImageSegmentation` will be deprecated: https://github.com/huggingface/transformers/blob/892399c5ffdc3c7d0e4f9b2c1c81511293c1581b/src/transformers/models/auto/modeling_auto.py#L487
So we'll have `AutoModelForSemanticSegmentation`, `AutoModelForUniversalSegmentation` and `AutoModelForInstanceSegmentation`.
Then, I don't know what the project regarding the `image-segmentation` pipeline task in Transformers is moving forward, I'm going to ask.

Comment (michaelbenayoun) [reply optimum/exporters/onnx/model_configs.py:576]: Alright. Do not want to complicate things, we can merge, but let's add a comment explaining this in the code so we understand in the future.

Comment (regisss) [reply optimum/exporters/onnx/model_configs.py:576]: Another solution is to remove the `"pred_boxes"` and `"pred_masks"` outputs of "image-segmentation": https://github.com/huggingface/optimum/blob/1a5443b8ed8cd4ce1375055817ae6e75926939a6/optimum/exporters/onnx/base.py#L136
I don't remember why I put them when I added this. But this won't be backwards compatible.

Comment (michaelbenayoun) [reply optimum/exporters/onnx/model_configs.py:576]: Are they useful for an `image-segmentation` task?
I think I am the one who added that comment, or at least we already had the discussionm dont remember lol!

**Review by fxmarty — APPROVED** [review]

File: optimum/exporters/onnx/model_configs.py (hunk 1 above)
Comment (regisss) [reply optimum/exporters/onnx/model_configs.py]: Only `pred_masks` is useful for segmentation. I just pushed a new commit to refine that.

**Review by michaelbenayoun — APPROVED** [review]
LGTM

**Pull request merged**